controllers/storageclassclaim_migration_controller.go

```diff
@@ -0,0 +1,117 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/red-hat-storage/ocs-client-operator/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	// for migration of storageclassclaims to storageclaims
+	storageClassClaimFinalizer = "storageclassclaim.ocs.openshift.io"
+)
+
+// StorageClassClaimReconcile migrates StorageClassClaim objects to StorageClaim
+type StorageClassClaimMigrationReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	log logr.Logger
+	ctx context.Context
+}
+
+func (r *StorageClassClaimMigrationReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&v1alpha1.StorageClassClaim{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Complete(r)
+}
+
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=ocs.openshift.io,resources=storageclassclaims/finalizers,verbs=update
+
+func (r *StorageClassClaimMigrationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	r.log = log.FromContext(ctx)
+	r.ctx = ctx
+	r.log.Info("Starting reconcile.")
+
+	storageClassClaim := &v1alpha1.StorageClassClaim{}
+	storageClassClaim.Name = req.Name
+
+	if err := r.get(storageClassClaim); err != nil && !kerrors.IsNotFound(err) {
+		r.log.Error(err, fmt.Sprintf("failed to get storageclassclaim %q", storageClassClaim.Name))
+		return ctrl.Result{}, err
+	}
+
+	storageClaim := &v1alpha1.StorageClaim{}
+	storageClaim.Name = storageClassClaim.Name
+
+	// blockpool (trim "pool") -> block and sharedfilesystem (trim "pool") -> sharefilesystem
+	storageClaim.Spec.Type = strings.TrimSuffix(storageClassClaim.Spec.Type, "pool")
```

Comment:
The logic does not hold, you are not translating `sharedfilesystem` into `sharedfile`
Instead of trimming we fix the issue by writing more verbose code:

```suggestion
	switch if storageClassClaim.Spec.Type {
		case "blockpool":
			storageClaim.Spec.Type = "block"
		case "sharedfilesystempool": 
			storageClaim.Spec.Type = "sharedfile"
	}
```

Reply:
> you are not translating
- yes I'm not as I don't know we want to rename `sharedfilesystem` to `sharedfile`.